features/consent/src/main/java/in/projecteka/jataayu/consent/ui/fragment/ConfirmPinFragment.kt

```diff
@@ -81,34 +81,40 @@ class ConfirmPinFragment : BaseDialogFragment(), OtpSubmissionClickHandler, OtpC
-                    viewModel.createPinResponse.observe(this, Observer {
+                    viewModel.createPinResponse.observe(this, Observer { payloadResource ->
 
-                        when (it) {
-                            is Loading -> showProgressBar(it.isLoading, getString(R.string.creating_pin))
+                        when (payloadResource) {
+                            is Loading -> showProgressBar(payloadResource.isLoading, getString(R.string.creating_pin))
                             is Success -> {
-                                activity?.let {
-                                    it.setPinCreated(true)
+                                activity?.let { fragmentActivity ->
+                                    fragmentActivity.setPinCreated(true)
                                     showProgressBar(true)
-                                    viewModel.userVerificationResponse.observe(this, Observer { userVerificationResponse ->
-                                        activity?.setConsentTempToken(userVerificationResponse.temporaryToken)
-                                        EventBus.getDefault().post(MessageEventType.USER_VERIFIED)
-                                        it.setResult(Activity.RESULT_OK)
-                                        it.finish()
-                                    })
+                                    viewModel.userVerificationResponse.observe(
+                                        this,
+                                        Observer { userVerificationResponse ->
+                                            userVerificationResponse?.temporaryToken?.let {
+                                                activity?.setConsentTempToken(it)
+                                                EventBus.getDefault().post(MessageEventType.USER_VERIFIED)
+                                                fragmentActivity.setResult(Activity.RESULT_OK)
+                                                fragmentActivity.finish()
+                                            }
+                                        })
                                     viewModel.verifyUser(pin, this)
 
                                 }
                             }
                             is PartialFailure -> {
-                                context?.showAlertDialog(getString(R.string.failure), it.error?.message,
-                                    getString(android.R.string.ok))
+                                context?.showAlertDialog(
+                                    getString(R.string.failure), payloadResource.error?.message,
+                                    getString(android.R.string.ok)
+                                )
                             }
                         }
                     })
-                    if (context?.getConsentPinCreationAPIintegrationStatus()!!){
+                    if (context?.getConsentPinCreationAPIintegrationStatus()!!) {
```

Comment:
can write this as if(context?.getConsentPinCreationAPIintegrationStatus() == true){...} to remove the force unwrapping

Reply:
`== true` why? It will anyway go inside if it's true. Instead we can extract out the variable outside so it is unwrapped only once. But it's just one occurrence.

Reply:
if you have a boolean that is nullable and you need to check it, rather than `!!` force unwrapping or using the `let` block, you can evaluate that against a true / false value. 

`context?.getConsentPinCreationAPIintegrationStatus()` gives a `Boolean?` because context is nullable here. if you don't want to force unwrap, you could write this as 
`context?.getConsentPinCreationAPIintegrationStatus()?.let { if(it){...}}` 
or
`if(context?.getConsentPinCreationAPIintegrationStatus() == true){...} `